### PR TITLE
Improve inline documentation of ContributionType class

### LINKED ISSUE #499: Improve ContributionType checking
### Problem Description

Disclaimer: This issue may be based superseded or incomplete knowledge.

AFIK the computation of the `ContributionType.GEOMTRY_CHANGE` is computation intensive. That's why, if you are not interested in that type, you should use the `OSMContribution.is(ContributionType contributionType)` method instead of `getContributionTypes()`.

Yet, this method cannot check for empty `ContributionType` and as https://github.com/GIScience/oshdb/issues/113 was unfortunately closed, I will still have to check for `GEOMTRY_CHANGE`  to filter empty contribs.

### Expected Solution

Add a method to check for empty `ContributionType`.

### PR DESCRIPTION
**Pull request by SlowMo24**
This PR starts working on #499 

First step: update the documentation.

### CODE REVIEW

File: oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/mappable/OSMContribution.java [hunk 1]
```diff
@@ -83,24 +83,18 @@ public interface OSMContribution extends OSHDBMapReducible, Comparable<OSMContri
   OSHEntity getOSHEntity();
 
   /**
-   * Checks if this contribution is of the given contribution type.
+   * Checks if this contribution is of the given contribution type,
```
Comment (tyrasd) [inline oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/mappable/OSMContribution.java:86]:
```suggestion
   * Checks if this contribution is of the given contribution type.
```

File: oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/mappable/OSMContribution.java [hunk 2]
```diff
@@ -83,24 +83,18 @@ public interface OSMContribution extends OSHDBMapReducible, Comparable<OSMContri
...
-   *
-   * <p>
-   * If this is a entity creation or deletion, the other flags are not set (even though one might
-   * argue that a just created object clearly has a different geometry than before, for example).
-   * </p>
-   *
-   * @return a set of modification type this contribution made on the underlying data
+   * i.e. checks if the given contribution type is in the
+   * {@code EnumSet<ContributionType>} returned by
+   * {@link #mgetContributionTypes() getContributionTypes}. This method is
```
Comment (tyrasd) [inline oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/mappable/OSMContribution.java:91]: is the extra `m` a typo?

```suggestion
   * {@link #getContributionTypes() getContributionTypes}. This method is
```

**Comment by tyrasd** [comment]
> Closes #499

Does it, though? :wink:

**Comment by SlowMo24** [comment]
> Does it, though? wink

not yet, your right. But the goal would be to do that in this PR, I guess. Updated the description.

File: oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/mappable/OSMContribution.java (hunk 1 above)
Comment (SlowMo24) [reply oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/mappable/OSMContribution.java]: ah, I forgot...

**Comment by SlowMo24** [comment]
after our discussion in the last meeting: This PR is WIP to solve #499 but it is done concerning the single documentation update it does. I will not be able to work on #499 so feel free to merge this and leave #499 open (my suggestion).

**Review by tyrasd — APPROVED** [review]

**Pull request merged**